Warrior Release Compatibility

ahcbb6 opened the pull request:
Hello,

The following set of patches, provide compatibility to build with the **warrior** branch from **Yocto**,
it is important to note that this upgrades the branch three releases ahead, from rocko-> sumo->thud -> warrior.

More information about what each of these upgrades bring by themselves can be found at:
[Sumo Release Notes](https://lists.yoctoproject.org/pipermail/yocto-announce/2018-May/000136.html)
[Thud Release Notes](https://lists.yoctoproject.org/pipermail/yocto-announce/2018-November/000147.html)
[Warrior Release Notes](https://lists.yoctoproject.org/pipermail/yocto-announce/2019-May/000158.html)

Initially (on the first pull request) this was only able to build the fb-openbmc-image along with DISTRO="openbmc-fb".

It wasn't very clear which BSPs were supported at the beginning but after I saw the first pull request I could figure out the way the Travis CI worked and what was being tested, I was able to imitate the Travis CI flow so now the Warrior branch compatibility has been extended to build on:

**BSPs/Layers (and their images) now compatible with warrior:**
**- wedge**
**- wedge100**
**- yosemite**
**- lightning**
**- cmm**
**- galaxy100**
**- fbtp**
**- fbttn**
**- fby2**

For obvious reasons this provides many improvements in the system, including many CVEs and fixes from packages such as openssl, busybox, etc.

The system now correctly builds and runs a **kernel version 5.0.7** (linux-yocto)

There are a couple of necessary changes related to python3 because of the upgrade of the manifest generation / python3 packaging system itself (which I personally did on oe-core) [8d94b9](https://git.yoctoproject.org/cgit/cgit.cgi/poky/commit/?id=8d94b9db221d1def42f091b991903faa2d1651ce) and [b677787](https://git.yoctoproject.org/cgit/cgit.cgi/poky/commit/?id=b6777878ff03c3e956386020a19d11c875c835ae)

initscripts required some changes as well since it should be getting the ip command from iproute2 and not from busybox as it was happening before, also rdepend on dig since it is being used on the hostname script.

DISTRO wise, there was some required changes to DISTRO_FEATURES due to changes in OE-Core.
[9d973d3](https://git.yoctoproject.org/cgit/cgit.cgi/poky/commit/?id=9d973d3b4ed8d2ce3c8997dc21603a0952e7ef0e)

**This also contains many fixes to several packages including:**
**- u-boot-2016.07**
**- fw-util**
**- bios-util**
**- ipmitool**
**- vboot-utils**
**- tpm-tools**
**- trousers**
**- u-boot-tools**
**- libkv**
**- ast-mdio**
**- openbmc-gpio**
**- opkg_utils**
**- dhcp**
**- bind**
**- initscripts**

I would be open to discuss each of these individually if the commit message on each of them is not clear, but discussing each of these in the cover letter is not feasible.

The only downside of these set of commits is that it contains a temporary fix, because, since the inclusion of security flags on Yocto on: [30f9a507](https://git.yoctoproject.org/cgit/cgit.cgi/poky/commit/?id=30f9a5072d94f6c383b1ef5d25448af39691dce0) and the introduction of GCC 8, a lot of more security warnings are shown, and when using -Wall this causes a error while building several packages, I decided to squash all these in a single commit, to force them all not to work if this commit is not included, this way we'll be forced to analyze each of this cases individually, but the correct fix for these individual cases should come from the source code itself.
More info at:
[GCC8 - Warnings](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html)

There are also a couple of bugs I found while developing these, but I believe it would be better to file them through here, since while I could fix them, I would not be able to make an informed decision on the correct/best way to do so (e.g bios-util LICENSE is most certainly wrong).

This set of commits also upgrades **linux-aspeed (1250 and 2520)** to work with **Linux 5.0.3**, along with some dtb/kernel-fitimage fixes

Since I don't have hardware available at the moment, I have tested these at runtime on qemuarmv5 (qemuarm was moved to qemuarmv5 on warrior [6acb45 ](https://git.yoctoproject.org/cgit/cgit.cgi/poky/commit/?id=6acb45106dafaf6895ad1fb086b6977c256518f6)) , since it uses the same architecture/tune (arm926ejs) as several machines within the layer such as: mavericks, galaxy, lightning, wedge, wedge100 and yosemite, although it is worth mentioning that qemuarm (v7) does work correctly as well.

AFAIC there could still be some runtime issues, e.g. /mnt/data isnt mounted, like it would when using real hardware (due to the openbmc-utils dependency)

Regards,
Alejandro

ahcbb6 committed:
initscripts: Fix RDEPENDS

The hostname script relies on functionality of programs
such as ip and dig, these come from iproute2 and bind-utils
respectively.

Its intersting to note that on previous versions of this ip
was being used from busybox, although it is specifically
specified that ip should be used from iproute2 for full
functionality

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
bind: Delete unnecessary bbappend

OE-Core uses python3 by default now, the bbappend for
bind is not necessary anymore.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
okpg_utils: fix python3 RDEPENDS

python3-textutils and python3-subprocess packages do not
exist anymore on OE-Core (they are now part of python3-core),
hence the RDEPENDS need to be removed to avoid errors during
parsing.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
python3: Fix symbolic link and packaging

The current symbolic link uses /usr/bin incorrectly,
since it should use bitbake compatible variables like
${bindir}.

Create symlink from python3 to python (relative) correctly
and fix FILES_${PN} to include binaries from the python3
package.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-wedge: Provide layer compatibility

OE-Core now requires to set the layer compatibility to
a certain relase.

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-python3: Provide layer compatibility

OE-Core now requires to set the layer compatibility to
a certain relase.

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-facebook: Provide layer compatibility

OE-Core now requires to set the layer compatibility to
a certain relase.

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-aspeed: Provide layer compatibility

OE-Core now requires to set the layer compatibility to
a certain relase.

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
sync_yocto: Sync warrior branch from yocto as well

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-openbmc: Provide layer compatibility

OE-Core now requires to set the layer compatibility to
a certain relase.

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
dhcp: Add warrior compatibiity

The patch for dhcp should be added to warrior release as well

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
packagegroup-openbmc-python3: Fix unexisting packages on RDEPENDS

The packages python3-argparse, python3-subprocess and python3-enum
do not exist anymore on OE-Core (they are now part of python3-core).

Delete them from RDEPENDS since they are unnecessary.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
libkv: Fix configure for new distutils class

The distutils3 class now calls setup.py on do_configure,
but since the libkv package does not have a setup.py it
causes an error.

Override the distutils_do_configure() task to avoid such
error and continue with the compilation/packaging.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
ast-mdio: Migrate to Python3

This change is a bit confusing since ast-mdio used to
work with python2, but the python package has been
blacklisted, and also the dependencies have changed
with the OE-Core releases.

The ast-mdio script relies on subprocess, time and
argparse to work properly, and these are all coming
from python3-core now.

Change RDEPENDS_ast-mdio to python3-core for the
reasons stated above, and also change the shebang
of the script to use python3 instead to avoid QA
errors regarding runtime dependencies while packaging.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
CFLAGS: Silence security and gcc8 warnings (temporarily) for several packages

Since the adoption of GCC 8 > in Yocto along with the
introduction of SECURITY_STRINGFORMAT on 30f9a507,
several errors appeared through several software
packages included in the DISTRO (these either used
to be warnings before the SECURITY_CFLAGS were
introduced or they were not even warnings before
GCC 8.

The GCC 8 enabled warnings are documented here:
https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html

Some of these errors show up like this:

-----------------
lldp_util.c: In function 'lldp_open':
lldp_util.c:150:5: error: format not a string literal _
    and no format arguments [-Werror=format-security]
snprintf(ifr.ifr_name, IFNAMSIZ, interface);
-----------------

-----------------
| log.c: In function 'obmc_log_by_prio':
| log.c:184:20: error: format not a string literal and no format
arguments [-Werror=format-security]
|     fprintf(stream, buf);
-----------------

While the correct way to fix this is to do it within the
source code of each package (they shouldn't be showing
warnings anyway), as a TEMPORARY fix these warnings have
been silenced for now, some of these include:

 -Wno-format-security
 -Wno-unused-result
 -Wno-format-truncation
 -Wno-format-overflow
 -Wno-implicit-function-declaration

On each package these were all silenced on a per-need
basis, meaning that all the other security flags should
still be enabled.

In only 3 packages the whole SECURITY_CFLAGS had to be
disabled:
- obmc-pal
- libpal
- me-util

While some of these may only actually be warnings, a few
could potentially mean security threats, hence one of
the reasons I decide to squash all silencing commits into
a single one is to make them either all work or not, to
force them to be analyzed on an case by case basis.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
u-boot-tools: Add dtc-native dependency

When IMAGE_FSTYPES contains cpio the image_cpio task is executed,
this causes images to fail showing the following error:

| DEBUG: Executing shell function do_image_cpio
| 151619 blocks
| 2 blocks
| sh: dtc: command not found
| mkimage: Can't open fit-wedge-<date>.itb.tmp: No such file or
  directory
| WARNING: temp/run.do_image_cpio. exit 255 from 'mkimage -f
  recipe-sysroot/etc/fit-wedge.its ${FIT_DESTINATION}'
| ERROR: Function failed: do_image_cpio

This is caused due to a missing dependency from the native
u-boot-tools package to dtc-native, which tries to use it but
it is unavailable.

Create a bbappend to the u-boot-tools package to inject this
runtime dependency which allows mkimage to run correctly
fixing the issue.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
linux-aspeed: Upgrade to kernel version 5.0.3 for 1250

Upgrade aspeed 1250 based MACHINES to latest available version of 5.0.3.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
images: Remove unnecessary IMAGE_FSTYPES assignment

All these images inherit the kernel_fitimage class
which adds cpio.lzma.u-boot to IMAGE_FSTYPES.

It is unnecesary to assign this variable on the image
recipes as well as it has no effect.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
libipmi: Change PREFERRED_VERSION to 0.2

The libipmi recipe has been upgraded to 0.2, which is the
only recipe available.

Change PREFERRED_VERSION on MACHINES that still preferred
to use 0.1 to avoid warnings.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
openbmc-gpio: Remove python3 RDEPENDS that dont exist

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
vboot-utils: Remove python3 RDEPENDS that dont exist

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
packagegroup-openbmc-python3: Remove python3 RDEPENDS that dont exist

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
u-boot: Fix compilation error due to incompatibility with dtc

DTC does not use an underscore for the LIBFDT_H and
LIBFDT_ENV_H variables, this causes an error since u-boot does.

Add a patch that removes the underscore to allow compilation
to succeed.

This patch is only required on the warrior release due to the
version of dtc upgrade.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
linux-aspeed: Upgrade to kernel version 5.0.3 for 2520

Upgrade aspeed-2520 based MACHINES to latest available
version of 5.0.3.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
trousers: Upgrade to 0.3.14

The current version of trousers is not compatible with
openssl versions > 1.1.0.

This has been patched upstream in trousers on commit:
05411ea
Rather than backporting the patch, upgrade the trousers
recipe to 0.3.14 which contains this patch already.
A few other changes required were:
 - Fixing the unpack function
 - Fixing the LICENSE
 - Removing the 07-read_data-not-inline patch

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
tpm-tools: Upgrade to 1.3.9.1

- Fixed LICENSE
- Fixed Unpack function
- Removed unnecessary patches

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
u-boot: Fix u-boot 2016_07 build with openssl 1.1.x

The OpenSSL API majorly changed after 1.1.x, some of which
broke the RSA struct along with other methods, this patch
adds compatiblity with newer versions of OpenSSL

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
vboot-utils: Fix configure for new distutils class

The distutils3 class now calls setup.py on do_configure,
but since the vboot-utils package does not have a setup.py it
causes an error.

Override the distutils_do_configure() task to avoid such
error and continue with the compilation/packaging.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
Upgrade ipmitool to use git repo instead

The current implementation of ipmitool v1.8.18 contains
arcaic code that is incompatible with newer C compilers,
the last release was on Oct 2016 and several fixes,
(around 150 commits) have been pushed since.

Build ipmitool using the git repo instead, to avoid
compilation errors.
The FB plugin implementation for obmc using a patch
was rebased to work on the latest revision

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
front-paneld: Add CFLAGS correctly on compilation

The current Makefile is setting CFLAGS using "="
which resets anything set from the build system.

Append to CFLAGS instead of reassigning the variable,
allowing the build system to set its own CFLAGS.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
bios-util: Fix runtime dependency to Python3

The recipe for bios-util is missing a runtime dependency
to python3 which is the interpreter that should run the
script.

Add python3-core to RDEPENDS to fix this at build time
and fix the shebang on the script itself to call the
correct interpreter

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
log-util-v2: Pass -Werror and -Wall on CFLAGS through the recipe

Instead of passing -Werror and -Wall from the Makefile,
we pass these on via the recipe and they end up on the
compilation anyway.

The reason for this change is that it allows the
buildsystem to control all CFLAGS, append and remove
them when required

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
kernel_fitimage: Remove KERNEL_IMAGETYPE from DTB_PATH

The build system no longer builds a KERNEL_IMAGETYPE-DTB
file, hence this now causes an error like the following:

------------
FATAL ERROR: Couldn't open
"deploy/images/cmm/zImage-aspeed-bmc-facebook-cmm.dtb":
No such file or directory

mkimage: Can't open
tmp/deploy/images/cmm/fit-cmm-20190707051202.itb.tmp:
No such file or directory

ERROR: Function failed: do_image_cpio
-------------

Remove KERNEL_IMAGETYPE from the DTB_PATH variable so
it finds the correct device tree blob and allows the
cpio_image to be built correctly.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-fbtp: Provide layer compatibility

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-fbttn: Provide layer compatibility

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-fby2: Provide layer compatibility

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-wedge100: Provide layer compatibility

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-yosemite: Provide layer compatibility

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-lightning: Provide layer compatibility

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-galaxy100: Provide layer compatibility

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-cmm: Provide layer compatibility

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 commented:
Well, I've been monitoring the build and an unrelated error was thrown, I believe the Travis CI infrastructure does not support changes like this, and it tries to protect itself from performing too many jobs.

This is what is shown (tail):
0: python3-3.7.2-r0 do_compile - 318s (pid 26594)
1: gmp-6.1.2-r0 do_compile - 75s (pid 5734)
[Currently  2 running tasks (782 of 3331)  23% |#######                         |
0: python3-3.7.2-r0 do_compile - 323s (pid 26594)
1: gmp-6.1.2-r0 do_compile - 80s (pid 5734)

The job exceeded the maximum time limit for jobs, and has been terminated.
[LOG](https://api.travis-ci.com/v3/job/213826508/log.txt)

This will certainly happen for all other MACHINES/BSPs as well.

I believe that its caused because usually, most of the Shared State is already available and its not necessary to perform this many jobs, in this case though, since you don't build for the warrior branch (and its two releases ahead) you don't have the required shared state already available, its basically trying to build most if not everything from scratch.

Is there any way I can make a special request in which the limit of number of jobs is modified?, I have already tested all the listed MACHINES and images myself, but I'm a bit disappointed by this, because I wanted you to see it working too, if possible, I'd like this to be tested on the Travis CI.

Thanks!

Alejandro

ahcbb6 commented:
Hello @amithash , @benwei13 , can you modify the Travis CI so the build is able to continue?, like I said I've already built everything by imitating what the Travis CI does myself, but I'd like to see it built on your infrastructure anyway.

Thanks!

Alejandro

amithash commented:
@aehs29 This is super cool!!! I agree, we probably should fix the compiler warnings. I am ok getting the temp cflags to disable said warnings and fixing them one by one and removing the exception.

For fw-util, Let me work on deleting the 0.1 version of fw-util so you don't have to copy-paste the entire source! :-( Once I do that you can rebase your tree.

PS: Our auto tooling is probably better at getting PRs one diff at a time.

ahcbb6 commented:
Awesome!, yeah absolutely, I can definitely rebase my tree on top of that, the only reason I didn't delete the fw-utils_0.1 recipe myself is because I didn't want to make a misinformed decision that would break someone else's flow (perhaps something/someone was still using it but I couldn't be sure), but yeah, it'd be a lot better if we have it in a common place.

I have a similar issue with the kernel_fitimage class (coming from meta-aspeed) vs kernel-fitimage class (coming from oe-core), in most situations I would rather use the one from oe-core, but the fact that it is named differently (looks like on purpose) makes me think there is a reason why we don't use the one from oe-core, I just, at this point don't know why, and decided to keep and fix the current one for now.

amithash commented on common/recipes-core/initscripts/files/hostname.sh:
```diff
@@ -17,7 +17,7 @@ set_hostname() {
   hostname $name
   # Reload/Restart any services needing the updated hostname
-  /etc/init.d/syslog reload
+  /etc/init.d/syslog restart
```
I believe we use rsyslog not syslog which does support reload:
https://github.com/facebook/openbmc/blob/helium/common/recipes-extended/rsyslog/files/initscript#L91

ahcbb6 replied:
True, I believe this might have been caused because of my runtime testing on qemuarmv5 (same TUNE) instead of real hardware

ahcbb6 replied:
I just checked and yes, the qemuarmv5 warrior build for fb-openbmc-image does not pull rsyslog (haven't checked why yet)

amithash commented on openbmc-init-build-env:
```diff
@@ -1,7 +1,7 @@
 
 if [ -z "$DISTRO_DEFAULT" ]; then
-  DISTRO_DEFAULT=rocko
+  DISTRO_DEFAULT=warrior
```
Oops. This switches everyone to warrier by default :-)
Generally we do full qualification on a per-platform basis before switching it over and only after all platforms have done their qualification, do we switch the default. I'd leave this as rocko for now but probably switch portwell to warrior and have it lead the effort

ahcbb6 replied:
Of course, the goal of this commit was only for the Travis CI to be able to test it automatically, but at least until we have something that we know for sure is more stable it should stay as rocko

amithash commented on meta-facebook/meta-python3/recipes-connectivity/bind/bind_%.bbappend:
```diff
@@ -1,3 +0,0 @@
-# Python3 only. Override py2 packages
-RDEPENDS_${PN}_remove = "python-core"
-RDEPENDS_${PN} += " python3-core"
```
Is that true for rocko as well?

ahcbb6 replied:
Yes, I do believe so.
https://git.yoctoproject.org/cgit/cgit.cgi/poky/tree/meta/recipes-connectivity/bind/bind_9.10.5-P3.bb?h=rocko

amithash commented on meta-facebook/meta-wedge100/conf/layer.conf:
```diff
@@ -8,3 +8,5 @@ BBFILES += "${LAYERDIR}/recipes-*/*/*.bb \
 BBFILE_PATTERN_wedge100 = "^${LAYERDIR}/"
 BBFILE_PRIORITY_wedge100 = "25"
+
+LAYERSERIES_COMPAT_wedge100 = "warrior"
```
Id change this to "warrier rocko"
Since we will continue to build rocko till we fully qualify the platform on the new layer and then deprecate the old.

ahcbb6 replied:
Sure, we can do that, the variable LAYERSERIES_COMPAT_<override> does exist in rocko as well, but the warnings/errors for building without it are not present yet, basically rocko does not require it to be set, but I do agree, we should specify both.

amithash commented on meta-facebook/conf/distro/openbmc-fb.conf:
```diff
@@ -132,9 +132,9 @@ DEFAULT_TIMEZONE ?= "America/Los_Angeles"
-OPENBMC_FB_DEFAULT_FEATURE = "ext2 ipv6 largefile usbgadget usbhost nfs zeroconf"
+OPENBMC_FB_DEFAULT_FEATURE = "ext2 ipv6 largefile usbgadget usbhost nfs zeroconf ipv4"
 OPENBMC_BOARD_FEATURE ?= ""
-DISTRO_FEATURES = "${DISTRO_FEATURES_LIBC} ${OPENBMC_FB_DEFAULT_FEATURE} ${OPENBMC_BOARD_FEATURE}"
```
Did this change happen pre-rocko or between rocko and warrior? I am trying to see if we can do it in some backwards compatible way.

ahcbb6 replied:
I think this happened somewhere in between thud-> warrior.

I think having it OPENBMC_FB_DEFAULT_FEATURE works out, we could still leave DISTRO_FEATURES_LIBC for compatibility, just dont rely on it, for example if we are certain we need ipv4,  it wont affect if someone has it on DISTRO_FEATURES_LIBC and you put it on OPENBMC_FB_DEFAULT_FEATURE as well.

Lets just not rely on it coming from default-distrovars.inc

amithash reviewed (commented):
I went ahead and deleted 0.1 version of fw-util from meta-yosemite, meta-fby2, meta-fbttn and meta-minilaketb. I am guessing we wont need this patch (and similar patches for the other platforms) anymore.

ahcbb6 commented:
Great, just for sanity, should we check the priorities set for BBFILE_PRIOIRITY for the different layers?, what is the reasoning behind the BBFILE_PRIORITY on some layers, for example fby2 has 8 and fbttn has 20, is this something we have to review to make sure its doing what we want?.

Thanks!
Alejandro

ahcbb6 committed:
openbmc-fb: Set DISTRO_FEATURES correctly

The variable $DISTRO_FEATURES_LIBC is no longer used in
OE-Core, this causes errors during boot since the system
was expecting ipv4 by default (busyboxs config changes
depending on features, and it affects things like ifup).

Add ipv4 to OPENBMC_FB_DEFAULT_FEATURE to fix the issue,
and only keep DISTRO_FEATURES_LIBC for compatibility,
but don't rely on it since it is not being set on
default-distrovars.inc on upstream anymore.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
simulate_travis_ci_build.sh: Temporary commit to test changes without Travis CI automatic build

Travis CI builds fail due to the amount of changes, so
create a simple script that naively simulates what Travis
is doing, for obvious reasons its slower since it builds
sequentially.

Usage examples:
$ BOARDS="wedge" ./simulate_travis_ci_build.sh
$ DISTRO_DEFAULT="rocko" BOARDS="wedge yosemite" ./simulate_travis_ci_build.sh
$ ./simulate_travis_ci_build.sh

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-facebook: Set warnings instead of fatal errors for missing bbappends

As the project supports more and more releases from Yocto,
the recipes vary more frequently, in some cases, bbappends
would fail to find a respective .bb file, the default
behavior for this is to cause an error, we have to set it
to show a warning instead to allow the cooker to proceed

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 commented:
Hey guys ( @amithash )

I finally had some time this weekend to play around with this again and solve the pending issues, sorry for the delay.

**Changes since the last pull request:**

**- Rebased on top of latest helium**

**- Removed the following commits due to our previous conversations:**
_fw-util: Port the fw-util and fw-util-test recipes to meta-fby2_
_fw-util: Port the fw-util and fw-util-test recipes to meta-fbttn_
These two were no longer necessary since the fw_util_0.1 recipe was
deleted on your part, thanks!.

_- initscripts: Fix syslog syntax on hostname._
This one was not necesary due to the usage of rsyslog instead.

**Resolved conflicts on:**
_- CFLAGS: Silence security and gcc8 warnings (temporarily) for several packages_
Due to the fw-util_0.1 not existing anymore

_- libipmi: Change PREFERRED_VERSION to 0.2_
libipmi PREFFERED_VERSION does not exist for cnm anymore

_- linux-aspeed: Upgrade to kernel version 5.0.3 for 1250_
This commit used to provide a defconfig for kernel v5, but a similar defconfig
has now been merged on helium

_- openbmc-fb: Set DISTRO_FEATURES correctly_
Package management was recently added on helium, which conflicted with the previous change.
This was rebased on top of that change, and also kept DISTRO_FEATURES_LIBC for compatibility
purposes from the conversation we had, but fixed the ipv4 issue that still remained.


_- u-boot: Fix compilation error due to incompatibility with dtc_
This commit had to change, due to our previous conversation on which this pull
request should work on both rocko AND warrior, the patch for dtc compatibility
is only required on warrior due to the upgrade in dtc version, so, fix this
by conditionally adding the patch depending on the release that is building.

As a side note, I still believe having separate branches for stable releases might
be easier, since as releases progress they fall apart from each other and will
eventually require more and more workarounds like this, but I do understand that
there might be reasons why we might to keep this way as well.


**- Changed LAYERCOMPAT to warrior AND rocko for all the previous layers where it was added**

**meta-facebook: Set warnings instead of fatal errors for missing bbappends**
- I added a commit that sets BB_DANGLINGAPPENDS_WARNONLY to change default behavior of
missing bb files to a respective bbappend, because the recipe u-boot-tools which requires a bbappend did not exist on the rocko release for example.
```
--------------------
ERROR: No recipes available for:
rocko/poky/../../../meta-aspeed/recipes-bsp/u-boot/u-boot-tools_%.bbappend
--------------------
```
Default behavior causes the cooker to exit with a fatal error, but setting it to show warnings
allows the build to keep going.
Currently I set this on meta-facebook layer.conf since its pretty global, otherwise it would
have to be set on each layer or local.conf.sample individually.

**Keep rocko as DEFAULT_DISTRO**
- I deleted the commit that set warrior as DEFAULT_DISTRO, again this change was because I thought there would be a separate branch for warrior itself, so, for now, rocko is kept as default.

**Added Testing script**
- I put a temporary commit at the top, on which a script is added to naively emulate
what Travis CI does (since we know this many changes breaks it), it basically builds for
the same default boards Travis does (more BOARDS could be added if necessary using the
BOARDS variable), but does it sequentially, downloads are shared amongst builds to speed up the process a little bit, but this obviously still takes a while.

Usage is very simple:
```
$ BOARDS="wedge" ./simulate_travis_ci_build.sh # builds wedge board on DISTRO_DEFAULT (rocko)
$ DISTRO_DEFAULT="warrior" BOARDS="yosemite fby2" ./simulate_travis_ci_build.sh  # Builds both yosemite and fby2 for warrior release
$ DISTRO_DEFAULT="warrior" ./simulate_travis_ci_build.sh # Builds all boards that are used as default on Travis CI for warrior release
```
- The Shared State Cache issue that I previously mentioned can be easily seen when building sequentially, as I said debugging the sstate shows this is caused because OPENBMC_VERSION invalidates tasks in openssh and that causes a chain reaction on all dependent tasks.

**- I've tested builds for all the Travis CI default boards both on rocko and warrior, from scratch and using the sstate cache as well.**

ahcbb6 commented:
Great I think the Travis CI passed (for the most part haha), Its getting an unrelated error when installing packages on the Ubuntu host:

[Build 266](https://travis-ci.com/facebook/openbmc/builds/121910657)
Failed on lightining:
```
apt-get install failed
The command "sudo -E apt-get -yq --no-install-suggests --no-install-recommends $(travis_apt_get_options) install build-essential wget diffstat chrpath texinfo bzip2 libthread-queue-perl" failed and exited with 100 during .
```

Even [build 265](https://travis-ci.com/facebook/openbmc/jobs/222356635) passed, but failed on wedge with a similar error:
apt-get install failed
`The command "sudo -E apt-get -yq --no-install-suggests --no-install-recommends $(travis_apt_get_options) install build-essential wget diffstat chrpath texinfo bzip2 libthread-queue-perl" failed and exited with 100 during .`

Juts pointing out that this did not break the Travis CI as before, since the DEFAULT_DISTRO is still set to rocko, so you would still need to use the provided script to test for warrior manually, but it does show that my warrior compatibility changes don't break rocko builds which is great!

Check it when you get a chance and let me know @amithash !

Thanks!
Alejandro

ahcbb6 committed:
meta-wedge400: Provide layer compatibility

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-fby3: Provide layer compatibility

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-yamp: Provide layer compatibility

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-minilaketb: Provide layer compatibility

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-minipack: Provide layer compatibility

Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 commented:
I added additional support for the following boards:
**- wedge400**
**- fby3**
**- yamp**
**- minilaketb**
**- minipack**

Even though these boards are not tested automatically by Travis

PD: Also, it looks like Travis is still having network issues

ahcbb6 committed:
uboot: Add bison dependency to uboot and uboot-fw-utils

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-fbal: Provide layer compatibility

    Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-ep: Provide layer compatibility

    Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 committed:
meta-sp: Provide layer compatibility

    Provide LAYERSERIES_COMPAT for warrior and rocko releases.

Signed-off-by: Alejandro Enedino Hernandez Samaniego <aehs29@gmail.com>

ahcbb6 commented:
Hello @amithash:

I added additional support for the following boards on the warrior branch:
**- fbal (angelslanding)**
**- fbep (emeraldspool)**
**- fbsp (sonorapass)**

Even though these boards are not tested automatically by Travis

As far as I'm concerned these are **ALL** the supported boards by OpenBMC.

At this point, after these commits can be merged, I'd like to have a conversation to see how we can proceed, the things that I'd like to do next is:
- Unify the recipes / clean up the meta layers (remove duplicated recipes on several layers and correctly use bbappends when required)
- Start fixing the compilation warnings for each individual components, starting with the ones where all the security warnings had to be disabled (e.g. obmc-pal, libpal, me-util)
- Fix sstate issue on OpenBMC version

ahcbb6 commented:
Oops, forgot to say that using the yocto-kernel-cache infrastructure for defining/creating kernel configurations for all the MACHINEs (vs defconfigs) should also be on the list, just commenting it here for bookkeeping.

ahcbb6 commented:
Rebased on top of helium

ahcbb6 commented:
Hey @amithash , could we close this pull request as soon as possible?, its hard to keep rebasing this many commits on top of helium for every change.

amithash commented:
@aehs29 I will be pulling in these patches one by one. The first one might close the PR but do not fear, the others are on the way.
